lnwallet/chanfunding/assembler.go

```diff
@@ -59,6 +59,11 @@ type Request struct {
 	// the funding output.
 	RemoteAmt btcutil.Amount
 
+	// FundUpToMaxAmt should be set to a non-zero amount if the channel
+	// funding should try to add as many funds to LocalAmt as possible
+	// until at most that amount is reached.
+	FundUpToMaxAmt btcutil.Amount
```

Comment:
after a full pass of the PR, I see we are using LocalAmt for min-allowed funding amount. Left a question on that part re if we should be re-purposing the LocalAmt for that or if we should not rather thread the min & max config values through to the assembler

Reply:
Yes, initially LocalAmt is set to the min-allowed amount and then evaluated against the remoteInitialBalance and other sanity checks in `rpcserver.go` along the way down to the assembler, so the code at this stage kinda expects the LocalAmt to be set(it's not permitted to set LocalAmt>0 and the `fundmax` in both `lncli` and the rpc). The `wallet_assembler.go` also expects the LocalAmt to be set - otherwise it considers the funding flow initiated by the remote end and we won't start selecting coins on our end(see https://github.com/lightningnetwork/lnd/blob/master/lnwallet/chanfunding/wallet_assembler.go#L288). So if we'd leave the LocalAmt blank we would have to fix its sanity checks if we'd introduce smth like a new `fundMax_min` parameter to the funding flow.

I think conceptually it also makes sense to re-use the LocalAmt as the other switch-cases in `wallet_assembler.go` in your comment below also use it to determine the `localContributionAmt`.

Reply:
Sounds good, will update the comment or change sanity checks depending on what the conclusion here will be.

Reply:
@ellemouton just FYI. I am currently working on integrating the `FundMinAmt` into the sanity checks but this is taking me longer than expected as I am trying to fix failing tests along the way.

Reply:
I've added a field `FundMinAmt` to represent the lower bound of a channel capacity and also adjusted the sanity checks along the way to the coin selection.

Reply:
Yes, all changed.